server.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+var nameMatcher = require("./lib/nameMatcher.js");
+var hapi = require('hapi');
+var fs = require('fs');
+nameMatcher.whenReady.then(function(value) {
+    var server = new hapi.Server();
+    server.connection({
+        tls: {
+            key: fs.readFileSync('./cert/self.key'),
+            cert: fs.readFileSync('./cert/self.crt')
+        },
+        port: 3000
+    });
+    // production match
+    server.route({
+        method: 'GET',
+        path: '/{names}',
+        handler: function (request, reply) {
+            var nameList = request.params.names.split(":");
+            var nameObjects  = nameList.map(function createNameObjects(current) {
+                var nameObj = {};
+                current = current.split(",");
+                nameObj.first = current[0];
+                nameObj.last = current[1];
+                
+                return nameObj;
+            });
+          
+            reply(nameMatcher.matchNames(nameObjects, "prod"));
+        }
+    });
+    // production add 
+    server.route({
+        method: 'POST',
+        path: '/',
+        handler: function (request, reply) {
+            if (request.payload.first !== undefined  
+                && request.payload.last !== undefined ){
+	        var firstList = request.payload.first.split(',');
+                var lastList = request.payload.last.split(',');
+                if (firstList.length !== lastList.length) {
+                    reply('failure: unpaired first/last');
```

Comment:
what process is consuming this service?  it may be **more work** to make the consuming process have to sniff success callbacks for failures then manually divert control in each case to a failHandler *than it would be* to simply pass an Error object! Pretty much all you have to do is `require('boom')` in, then reply with an error with the text you already have!  http://hapijs.com/api#error-response.  `return reply(Boom.badRequest('j-winky didnt match nuthin'));`